### PR TITLE
feat(newserver): Medal of Honor: Allied Assault

### LINKED ISSUE #2393: [Server Request] Medal of Honor: Allied Assault
## What game server would you like adding?

Medal of Honor: Allied Assault

## Can you link to any tutorials or guides?

http://www.uhost4free.com/mohaa.htm

## Is the server on Steam?

* [ ] Yes
* [X] No

## Is the server supported on Linux?

We only support Linux servers and do not support Wine.

* [X] Yes
* [ ] No

### PR DESCRIPTION
**Pull request by Frisasky**
# Description

check discord message for file download url

Logo: https://gepig.com/game_cover_460w/495.jpg

Fixes #2393

## Type of change

* [ ] Bug fix (change which fixes an issue).
* [ ] New feature (change which adds functionality).
* [x] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).
* [ ] This change requires a documentation update.

## Checklist

* [ ] This code follows the style guidelines of this project.
* [x] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [ ] I have provided Co-author details below.
* [ ] I have performed a self-review of my own code.
* [ ] I have squashed commits.
* [ ] I have commented my code, particularly in hard to understand areas.
* [ ] I have made corresponding changes to the documentation if required.

## Provide Github Email

Fill out below info or tick box below:
```
Co-authored-by: John Doe <name@example.com>
```

- [ ] I do not wish to provide an email. I am aware this will hide me as the author of this commit.

All pull requests will now be squashed to create a tidy commit history and simplify changelog creation. You can provide either your own email or a GitHub-provided no-reply email.

When a PR is squashed the author becomes the person who squashed the PR. This removes you as the author of your own PR.
The only workaround for this is to add your details as a co-author. More info about co-authors can be found [here](https://help.github.com/en/articles/creating-a-commit-with-multiple-authors).

### CODE REVIEW

File: lgsm/functions/check_deps.sh [hunk 1]
```diff
@@ -397,6 +397,9 @@ fn_deps_build_debian(){
 		else
 			array_deps_required+=( openjdk-8-jre-headless )
 		fi
+	# Medal of Honor: Allied Assault 
+	if [ "${shortname}" == "mohaa" ]; then
```
Comment (h3o66) [inline lgsm/functions/check_deps.sh]: i guess this should be a elif instead of a if

**Comment by Frisasky** [comment]
the server is now ready for testing @dgibbs64

**Comment by dgibbs64** [comment]
I am getting the following error
```
--- Common Initialization ---
Medal of Honor Allied Assault 1.11 linux-i386 Jul 22 2004
----- FS_Startup -----
Current search path:
/home/dgibbs/.mohaa/main
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/pak6.pk3 (104 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak5.pk3 (259 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak4.pk3 (593 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak3.pk3 (669 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak2.pk3 (4722 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak1.pk3 (772 files)
/home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak0.pk3 (11175 files)

----------------------
18294 files in pk3 files
execing default.cfg
execing menu.cfg
execing newconfig.cfg
Config: unnamedsoldier.cfg
execing configs/unnamedsoldier.cfg
couldn't exec localized.cfg
execing autoexec.cfg
You are now setup for easy mode.
Opening IP socket: 192.168.0.240:12203
Hostname: pop-os.localdomain
Alias: pop-os
IP: 127.0.1.1
--- Common Initialization Complete --- 234 ms
--- Localization: I see 0 localization files
--- Localization: reading file global/localization.txt
Loading Localization File global/localization.txt
Loaded 1242 localization entries
couldn't exec mohaaserver.cfg
Can't find map maps/mohdm1.bsp
```
Looks like the map is missing?

**Comment by Frisasky** [comment]
> 
> 
> I am getting the following error
> 
> ```
> --- Common Initialization ---
> Medal of Honor Allied Assault 1.11 linux-i386 Jul 22 2004
> ----- FS_Startup -----
> Current search path:
> /home/dgibbs/.mohaa/main
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/pak6.pk3 (104 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak5.pk3 (259 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak4.pk3 (593 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak3.pk3 (669 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak2.pk3 (4722 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak1.pk3 (772 files)
> /home/dgibbs/linuxgsm/mohaaserver/serverfiles/main/Pak0.pk3 (11175 files)
> 
> ----------------------
> 18294 files in pk3 files
> execing default.cfg
> execing menu.cfg
> execing newconfig.cfg
> Config: unnamedsoldier.cfg
> execing configs/unnamedsoldier.cfg
> couldn't exec localized.cfg
> execing autoexec.cfg
> You are now setup for easy mode.
> Opening IP socket: 192.168.0.240:12203
> Hostname: pop-os.localdomain
> Alias: pop-os
> IP: 127.0.1.1
> --- Common Initialization Complete --- 234 ms
> --- Localization: I see 0 localization files
> --- Localization: reading file global/localization.txt
> Loading Localization File global/localization.txt
> Loaded 1242 localization entries
> couldn't exec mohaaserver.cfg
> Can't find map maps/mohdm1.bsp
> ```
> 
> Looks like the map is missing?

it should be fixed now

**Pull request merged**